PR check list for the dev_guide (#6346)
sunpy/sunpy: docs/dev_guide/contents/cheatsheet.rst

.. _cheatsheet:

******************************
SunPy Pull Request Cheat Sheet
******************************

The pull request (PR) cheat sheet below is a rough outline of the steps that should be taken when making a contribution to SunPy on Github.

> 

Reviewer: ```suggestion
New to Git and Github?
Check out the :ref:`newcomers` and `Git Cheat Sheets <https://training.github.com/>`__.

If you would prefer a visual Git interface, you can try `Github Desktop <https://desktop.github.com/>`__ or  `GitKraken <https://www.gitkraken.com/>`__.

```

    a. Resolve any conflicts that occur
4. Create a pull request
    a. `Create a pull request <https://docs.github.com/en/get-started/quickstart/hello-world#opening-a-pull-request>`_ from the branch you have updated
    b. Add a description to the PR that describes the changes you have made
    c. `Link <https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/autolinked-references-and-urls>`_ to any relevant Issues or PRs in your description
5. Add a changelog to your PR
    a. A `changelog <https://github.com/sunpy/sunpy/tree/main/changelog#changelog>`_ is a short record of the type of changes made in your PR
6. SunPy maintainers will `review your PR <https://docs.sunpy.org/en/latest/dev_guide/contents/pr_review_procedure.html>`_
    a. Fix any errors that others highlight and push the changes to your branch
    b. Discuss possible changes/improvements in the comments
    c. Review the `Continuous Integration (CI) <https://docs.sunpy.org/en/latest/dev_guide/contents/ci_jobs.html>`_ tests and fix any errors that are found
        i. If you are confused by an error that the CI is giving you just ask about in your PR

> 7. Ask questions if you get stuck or confused at any point

Reviewer: ```suggestion
#. Ask questions if you get stuck or confused at any point!
```

>     a. Open-source projects are about communication and collaboration

Reviewer: ```suggestion
    #. Open-source projects are about communication and collaboration.
```

>     b. Join the SunPy Matrix Chat channel (link at the bottom of this page)

Reviewer: ```suggestion
    b. `Join the SunPy Matrix Chat channel <https://matrix.to/#/+sunpy:openastronomy.org>`__.
```

> Making a change to the SunPy website? See `these guidelines <https://github.com/sunpy/sunpy.org#sunpyorg-website>`_ for making website changes.

Reviewer: ```suggestion
```
I think this is ok not being there.

> New to Git/Github? Check out the `Newcomers' Guide <https://docs.sunpy.org/en/latest/dev_guide/contents/newcomers.html>`_ and `Git Cheat Sheets <https://training.github.com/>`_.

Reviewer: ```suggestion
```
I am going to move this to the top.
